### PR TITLE
Add ademe site

### PR DESCRIPTION
**Pull request by sveetch**
This adds a new site for Ademe.

- [x] Create new site entry from a previous one
- [x] Apply theme colors
- [x] Apply specific assets
- [x] Customize templates if needed

### CODE REVIEW

File: sites/ademe/src/backend/ademe/settings.py [hunk 1]
```diff
@@ -0,0 +1,755 @@
...
+    EDX_USER_PROFILE_TO_DJANGO = {
+        "preferred_username": "username",
+        "email": "email",
+        "name": "full_name",
+        "given_name": "first_name",
+        "family_name": "last_name",
+        "locale": "language",
+        "user_id": "user_id",
+        "administrator": "is_staff",
+    }
```
Comment (sampaccoud) [inline sites/ademe/src/backend/ademe/settings.py:320]: you can delete that

File: sites/ademe/src/backend/ademe/settings.py [hunk 2]
```diff
@@ -0,0 +1,755 @@
...
+        "django.contrib.admin",
+        "django.contrib.sites",
+        "django.contrib.sitemaps",
+        "django.contrib.staticfiles",
+        "django.contrib.messages",
+    )
+
+    # Languages
+    # - Django
+    LANGUAGE_CODE = "en"
```
Comment (sampaccoud) [inline sites/ademe/src/backend/ademe/settings.py:431]: It should be `fr` by default like on `funmooc` for example.
```suggestion
    LANGUAGE_CODE = "fr"
```

File: sites/ademe/src/backend/ademe/settings.py [hunk 3]
```diff
@@ -0,0 +1,755 @@
...
+
+    # Careful! Languages should be ordered by priority, as this tuple is used to get
+    # fallback/default languages throughout the app.
+    # Use "en" as default as it is the language that is most likely to be spoken by any visitor
+    # when their preferred language, whatever it is, is unavailable
+    LANGUAGES = (("en", _("English")), ("fr", _("French")))
```
Comment (sampaccoud) [inline sites/ademe/src/backend/ademe/settings.py:437]:
```suggestion
    LANGUAGES = (("fr", _("French")), ("en", _("English")))
```

File: sites/ademe/src/backend/ademe/settings.py [hunk 4]
```diff
@@ -0,0 +1,755 @@
...
+                "level": "ERROR",
+                "handlers": ["console"],
+                "propagate": False,
+            }
+        },
+    }
+
+    # Demo
+    RICHIE_DEMO_SITE_DOMAIN = values.Value(
+        default="localhost:8080", environ_name="RICHIE_DEMO_SITE_DOMAIN", environ_prefix=None
```
Comment (sampaccoud) [inline sites/ademe/src/backend/ademe/settings.py:511]: When PR https://github.com/openfun/richie-site-factory/pull/274 is merged and you rebase, it should be port 8000 + other modifications as done in pointed PR for other sites.

File: sites/ademe/src/backend/ademe/wsgi.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+For more information on this file, see
+https://docs.djangoproject.com/en/1.8/howto/deployment/wsgi/
+"""
+
+import os
+
+from configurations.wsgi import get_wsgi_application
+
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "funmooc.settings")
```
Comment (sampaccoud) [inline sites/ademe/src/backend/ademe/wsgi.py:14]:
```suggestion
os.environ.setdefault("DJANGO_SETTINGS_MODULE", "ademe.settings")
```

File: sites/ademe/src/backend/templates/richie/styleguide/index.html [hunk 6]
```diff
@@ -0,0 +1,331 @@
+{% extends "richie/base.html" %}
```
Comment (sampaccoud) [inline sites/ademe/src/backend/templates/richie/styleguide/index.html:1]: It's a pity that we need to override this whole template :scream:  Can you explain why you override it and what we can change in richie to allow minimal overrides?

File: sites/ademe/src/frontend/scss/components/templates/courses/cms/_homepage.scss [hunk 7]
```diff
@@ -0,0 +1,41 @@
+@import 'richie-education/scss/components/templates/courses/cms/homepage';
+
+// Homepage customize
+//
```
Comment (sampaccoud) [inline sites/ademe/src/frontend/scss/components/templates/courses/cms/_homepage.scss:4]: can you explain why this override?

File: sites/ademe/src/frontend/scss/components/_subheader.scss [hunk 8]
```diff
@@ -0,0 +1,12 @@
+@import 'richie-education/scss/components/subheader';
+
```
Comment (sampaccoud) [inline sites/ademe/src/frontend/scss/components/_subheader.scss:2]: Can you explain why this override?

File: sites/ademe/src/backend/ademe/settings.py (hunk 4 above)
Comment (sveetch) [reply sites/ademe/src/backend/ademe/settings.py:511]: So i wait for this PR merge and i will just have to change 8080 to 8000 ?

File: sites/ademe/src/backend/templates/richie/styleguide/index.html (hunk 6 above)
Comment (sveetch) [reply sites/ademe/src/backend/templates/richie/styleguide/index.html:1]: Oh yeah, don't bother of it, this a temporary change for some needed debugging information with new color palette for Ademe. It was planned to be removed once i finished (which i realize it should be now).

There is definitively no reason for sites from factory to override the styleguide template.

File: sites/ademe/src/backend/ademe/settings.py (hunk 4 above)
Comment (sampaccoud) [reply sites/ademe/src/backend/ademe/settings.py:511]: It is now merged. You need to change the port and also do all the modifications that this PR is doing to all projects.

File: sites/ademe/src/backend/ademe/__init__.py [hunk 9]
```diff
@@ -0,0 +1 @@
+__version__ = "1.12.1"
```
Comment (sampaccoud) [inline sites/ademe/src/backend/ademe/__init__.py:1]:
```suggestion
__version__ = "0.0.1"
```

File: sites/ademe/src/backend/ademe/settings.py [hunk 10]
```diff
@@ -0,0 +1,719 @@
...
+
+    AUTHENTICATION_BACKENDS = ("django.contrib.auth.backends.ModelBackend",)
+
+    # AUTHENTICATION DELEGATION
+    RICHIE_AUTHENTICATION_DELEGATION = {
+        "BASE_URL": values.Value(
+            "", environ_name="AUTHENTICATION_BASE_URL", environ_prefix=None
+        ),
+        "BACKEND": values.Value(
+            "openedx-hawthorn",
```
Comment (sampaccoud) [inline sites/ademe/src/backend/ademe/settings.py:210]: @jbpenrath Ah

File: sites/ademe/src/backend/ademe/settings.py [hunk 11]
```diff
@@ -0,0 +1,719 @@
...
+    # LMS
+    RICHIE_LMS_BACKENDS = [
+        {
+            "BACKEND": values.Value(
+                "richie.apps.courses.lms.edx.EdXLMSBackend",
+                environ_name="EDX_BACKEND",
+                environ_prefix=None,
+            ),
+            "JS_BACKEND": values.Value(
+                "openedx-hawthorn",
```
Comment (sampaccoud) [inline sites/ademe/src/backend/ademe/settings.py:248]: @jbpenrath Ah

File: sites/ademe/src/backend/base/tests/test_cache.py [hunk 12]
```diff
@@ -0,0 +1,159 @@
+"""Test funmooc cache plugin"""
```
Comment (sampaccoud) [inline sites/ademe/src/backend/base/tests/test_cache.py:1]:
```suggestion
"""Test ademe cache plugin"""
```

File: sites/ademe/src/backend/ademe/settings.py (hunk 11 above)
Comment (sampaccoud) [reply sites/ademe/src/backend/ademe/settings.py:248]: @sveetch so we must put 
```suggestion
                "openedx-dogwood",
```

File: sites/ademe/src/backend/ademe/settings.py (hunk 10 above)
Comment (sampaccoud) [reply sites/ademe/src/backend/ademe/settings.py:210]: @sveetch so we must put
```suggestion
            "openedx-dogwod",
```

Comment (jbpenrath) [reply sites/ademe/src/backend/ademe/settings.py:210]: I named the backend with the oldest openedx version supported : 
So `openedx-hawthorn` supports hawthorn, ironwood, juniper and koa (tested on hawthorn, and ironwood, not quit sure for koa and never tested on juniper)
And `openedx-dogwood` supports Dogwood, Eucalyptus. (tested on both)

**Review by sampaccoud — APPROVED** [review]

**Pull request merged**